site/docs/api/sources-sinks.md

```diff
@@ -948,6 +948,99 @@ tests](https://github.com/hazelcast/hazelcast-jet-contrib/tree/master/xa-test)
 to get more information. This only applies to the JDBC sink, the source
 doesn't use XA transactions.
 
+### Elasticsearch
+
+Elasticsearch is a popular fulltext search engine.
+Hazelcast Jet can use it both as a source and a sink.
+
+#### Source
+
+The Elasticsearch connector source provides a builder and several 
+convenience factory methods.
+Most commonly on needs to provide
+- a client supplier, which returns a configured instance of
+RestHighLevelClient (see
+[Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high.html)), 
+- a search request supplier specifying a query to Elasticsearch,
+- a mapping function from `SearchHit` to a desired type.
+
+Example using a factory method:
+
+```java
+BatchSource<String> source = ElasticsearchSources.elasticsearch(
+    () -> client("user", "password", "host", 9200),
+    () -> new SearchRequest("my-index"),
+    hit -> (String) hit.getSourceAsMap().get("name")
+);
+```
+
+For all configuration options use the builder:
+
+```java
+BatchSource<String> elasticsearch = new ElasticsearchSourceBuilder<String>()
+        .name("elastic-source")
+        .clientSupplier(() -> new RestHighLevelClient(RestClient.builder(new HttpHost(
+                "localhost", 9200
+        ))))
+        .destroyFn(RestHighLevelClient::close)
+        .searchRequestSupplier(() -> new SearchRequest("my-index"))
+        .optionsFn(request -> RequestOptions.DEFAULT)
+        .mapHitFn(hit -> hit.getSourceAsString())
+        .slicing(true)
+        .build();
+```
+
+By default the connector uses a single scroll to read data from 
+Elasticsearch - there is only a single reader on a single node in whole 
+cluster.
+
+Slicing can be used to parallelize reading from an index with more
+shards. Number of slices equals to globalParallelism.
+
+If Hazelcast Jet nodes and Elasticsearch nodes are located on the same
+machines then the connector will use co-located reading, avoiding the
+overhead of physical network.
+
+#### Sink
+
+The Elasticsearch connector sink provides a builder and several 
+convenience factory methods.
+Most commonly you need to provide
```

Comment:
you need line breaks if it's intended to be another paragraph, otherwise you can merge it with previous line.